### PR TITLE
MO: deploy script

### PR DESCRIPTION
**Pull request by xavierlefevre**

### CODE REVIEW

**Comment by bam-bot-tech** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  **<a href="https://github.com/bamlab/dev-standards/blob/mo/deploy-script/code-quality/components-state-testing.mo.md">components-state-testing.mo.md</a>**: Seems you do not need a *Troubleshoot* part
  </td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>

  **<a href="https://github.com/bamlab/dev-standards/blob/mo/deploy-script/react-native/setup/deploy-script.mo.md">deploy-script.mo.md</a>**: MO doesn't contain a *Prerequisites* part are you sure your brain is all you need?
  </td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>

  **<a href="https://github.com/bamlab/dev-standards/blob/mo/deploy-script/react-native/setup/deploy-script.mo.md">deploy-script.mo.md</a>**: Seems you do not need a *Troubleshoot* part
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: react-native/setup/deploy.sh [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+  if [ $APP_ENV == "production" ]; then
+    echo "- - - - -"
+    echo "No Codepush for Production yet"
+    echo "- - - - -"
+  else
+    if [[ $APP_OS != "android" ]]; then
+      echo "- - - - -"
+      echo "Codepush iOS Staging"
+      echo "- - - - -"
+      code-push release-react -d Staging investo-ios ios -m --targetBinaryVersion $IOS_VERSION --des "$MESSAGE"
```
Comment (yleflour) [inline react-native/setup/deploy.sh]: project name, please commit --amend

File: react-native/setup/deploy.sh [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    fi
+    if [[ $APP_OS != "ios" ]]; then
+      echo "- - - - -"
+      echo "Codepush Android Staging"
+      echo "- - - - -"
+      code-push release-react -d Staging investo-android android -m --targetBinaryVersion $ANDROID_VERSION_NAME --des "$MESSAGE"
```
Comment (yleflour) [inline react-native/setup/deploy.sh]: same

File: react-native/setup/deploy-script.mo.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
+# [MO] Add a deploy script to your app
```
Comment (Minishlink) [inline react-native/setup/deploy-script.mo.md]: Required time is missing

File: react-native/setup/deploy.sh [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+APP_ENV=${APP_ENV:-"staging"}
+APP_OS=${APP_OS:-"ios and android"}
+DEPLOY_TYPE=${DEPLOY_TYPE:-"soft"}
+
+if [ $DEPLOY_TYPE == "hard" ]; then
+  echo "${BLUE}* * * * *"
+  echo "👷  Hard-Deploy"
+  echo "* * * * *${NO_COLOR}"
+  read -r -p "Did you bump the versions in fastlane/.env [y/N] " response
```
Comment (Minishlink) [inline react-native/setup/deploy.sh]: Only required if native code has changed ;)

File: react-native/setup/deploy.sh [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+  if [ $APP_ENV == "production" ]; then
+    echo "- - - - -"
+    echo "️️️⚠️  No Codepush for Production yet"
+    echo "- - - - -"
+  else
+    if [[ $APP_OS != "android" ]]; then
+      echo "${GREEN}- - - - -"
+      echo "Codepush 🍎  iOS Staging"
+      echo "- - - - -${NO_COLOR}"
+      code-push release-react -d Staging investo-ios ios -m --targetBinaryVersion $IOS_VERSION --des "$MESSAGE"
```
Comment (Minishlink) [inline react-native/setup/deploy.sh]: Project name should be in a env var or as a placeholder :o (this repo is public)

**Review by Minishlink — COMMENTED** [review]
There's some more goodies with this script 
(mandatory flag, forcing deployment/prevent deployment if Git repo is not clean, default changelog that can be changed) https://github.com/Minishlink/DailyScrum/blob/staging/bin/code-push.sh

**Review by yleflour — CHANGES_REQUESTED** [review]

**Comment by xavierlefevre** [comment]
@yleflour @Minishlink : pourriez rechecker ?

@Minishlink : j'ai pris la question "What's the changelog?" et j'ai enlevé : "Did you bump?" pour le moment

**Pull request merged**